parcels/nodes/PyNode.py

```diff
@@ -0,0 +1,601 @@
+import ctypes
+import sys
+import os
+from parcels.compilation import InterfaceC, GNUCompiler_SS, GNUCompiler_MS  # noqa: F401
+from parcels.tools import get_cache_dir, get_package_dir
+
+LIB_LOAD_MAX_REPEAT = 10
+
+# ======================================================================================================= #
+# filename "PyNode.py" is given because the wrap-compilation of "node.c" and "node.h" will result in      #
+# an auto-generated "node.py", which would then clash with this manually-defined superclass that uses it. #
+# ======================================================================================================= #
+
+
+class Node(object):
+    """
+    A simple nodal object, storing its connectivity to its predecessor and its successor, and (a reference to) its object (as payload)
+    """
+    prev = None
+    next = None
+    idgen = None
+    data = None
+    registered = False
+
+    def __init__(self, prev=None, next=None, id=None, data=None, c_lib_register=None, idgen=None):
+        """
+        Node - Constructor
+        :arg prev: predecessor Node object
+        :arg next: successor Node object
+        :arg id: optional, legacy parameter to set this object's ID
+        :arg data: data payload of this Node
+        :arg c_lib_register: a LibraryRegisterC object; available for compliance reasons with the NodeJIT sibling class
+        :arg idgen: an ID generator object used for ID registration- and de-registration for the Particle data payload
+        """
+        if prev is not None:
+            assert (isinstance(prev, Node))
+            self.set_prev(prev)
+        else:
+            self.reset_prev()
+        if next is not None:
+            assert (isinstance(next, Node))
+            self.set_next(next)
+        else:
+            self.reset_next()
+        if data is not None:
+            self.set_data(data)
+        else:
+            self.reset_data()
+        self.link()
+
+        assert idgen is not None, "Using Nodes requires to specify an ID generator (in order to release the ID on delete)."
```

Comment:
Provide (a link to) an example of such an ID generator here, so users are helped more if they encounter this problem?